### PR TITLE
HBASE-24416 RegionNormalizer spliting region should not be limited by…

### PR DESCRIPTION
**Pull request by ddupg**
… hbase.normalizer.min.region.count

JIRA: https://issues.apache.org/jira/browse/HBASE-24416#

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   4m  1s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-2.2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 39s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   1m  0s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   1m 30s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   4m 24s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 36s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   3m 45s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 42s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 16s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 27s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m 25s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  28m 11s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  javadoc  |   0m 34s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   3m 40s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 307m 26s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 37s |  The patch does not generate ASF License warnings.  |
|  |   | 380m 14s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.master.TestAssignmentManagerMetrics |
|   | hadoop.hbase.replication.multiwal.TestReplicationSyncUpToolWithMultipleAsyncWAL |
|   | hadoop.hbase.client.TestFromClientSideWithCoprocessor |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectoryWithRegionReplicas |
|   | hadoop.hbase.client.TestFromClientSide |
|   | hadoop.hbase.client.TestFromClientSide3 |
|   | hadoop.hbase.regionserver.TestSplitTransactionOnCluster |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.9 Server=19.03.9 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1788/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1788 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 5863c95dddb5 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1788/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / abe5a05bd1 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1788/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1788/1/testReport/ |
| Max. process+thread count | 5082 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1788/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by virajjasani** [comment]
@ddupg multiple tests are failing. Can you please take a look once? How are they doing locally?

**Comment by ddupg** [comment]
> @ddupg multiple tests are failing. Can you please take a look once? How are they doing locally?

Thanks @virajjasani for taking a look here. The following is unit log, it seems like that failed UTs are unrelated with this PR.

> [ERROR] ExecutionException The forked VM terminated without properly saying goodbye. VM crash or System.exit called?

Let me trigger UTs running again.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 36s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-2.2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 14s |  branch-2.2 passed  |
| +1 :green_heart: |  compile  |   0m 56s |  branch-2.2 passed  |
| +1 :green_heart: |  checkstyle  |   1m 20s |  branch-2.2 passed  |
| +1 :green_heart: |  shadedjars  |   4m  9s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  branch-2.2 passed  |
| +0 :ok: |  spotbugs  |   3m 19s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 17s |  branch-2.2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 48s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 17s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   4m  3s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  25m 27s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 2.10.0 or 3.1.2 3.2.1.  |
| +1 :green_heart: |  javadoc  |   0m 34s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   3m 14s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 163m 50s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 36s |  The patch does not generate ASF License warnings.  |
|  |   | 226m 55s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.TestZooKeeper |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.9 Server=19.03.9 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1788/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1788 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux bfaa4b984eb0 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1788/out/precommit/personality/provided.sh |
| git revision | branch-2.2 / 482956e532 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1788/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1788/2/testReport/ |
| Max. process+thread count | 4667 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1788/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/normalizer/SimpleRegionNormalizer.java [hunk 1]
```diff
@@ -148,14 +156,17 @@ public int compare(NormalizationPlan plan1, NormalizationPlan plan2) {
...
-      LOG.debug("Table " + table + " has " + nrRegions + " regions, required min number"
-        + " of regions for normalizer to run is " + MIN_REGION_COUNT + ", not running normalizer");
+    if (tableRegions == null) {
       return null;
     }
 
+    if (tableRegions.size() < minRegionCount) {
+      LOG.debug("Table " + table + " has " + tableRegions.size() + " regions, required min number"
+          + " of regions for normalizer merging to run is " + minRegionCount
+          + ", not running normalizer merging");
```
Comment (virajjasani) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/normalizer/SimpleRegionNormalizer.java:166]: Good to use parameters with `{}` for variables.

**Pull request merged**